lib/topic_query.rb

```diff
@@ -390,7 +390,7 @@ def prioritize_pinned_topics(topics, options)
     end
 
     unpinned_topics = topics.where("NOT ( #{pinned_clause} )")
-    pinned_topics = topics.dup.offset(nil).where(pinned_clause)
+    pinned_topics = topics.dup.offset(nil).where(pinned_clause).unscope(:order).order(pinned_at: :desc)
```

Comment:
Would `reorder()` work instead of `unscope().order()`?

Reply:
Yep!